src/main/java/org/jooq/lambda/PeekingIterator.java

```diff
@@ -0,0 +1,13 @@
+package org.jooq.lambda;
+
+import java.util.Iterator;
+
+public interface PeekingIterator<E> extends Iterator<E> {
```

Comment:
In fact, I wonder if this needs to be an interface, given we only have a single implementation so far...